lua/lualine/component.lua

```diff
@@ -274,14 +283,22 @@ function M:draw(default_highlight, is_focused)
   if self.options.fmt then
     status = self.options.fmt(status or '')
   end
-  if type(status) == 'string' and #status > 0 then
-    self.status = status
-    self:apply_icon()
-    self:apply_padding()
-    self:apply_on_click()
-    self:apply_highlights(default_highlight)
-    self:apply_section_separators()
-    self:apply_separator()
+  local show_separators_if_empty = self.options.show_separators_if_empty
+  if type(show_separators_if_empty) == 'function' then
+    show_separators_if_empty = show_separators_if_empty()
+  end
+  if type(status) == 'string' then
+    if #status > 0 or show_separators_if_empty then
+      self.status = status
+      if #status > 0 then
```

Comment:
We checked the status is larger then 0 while entering this block so this condition is redundent.

Reply:
If you're talking about line 293, the condition is not redundant since in 291 we tested if the length is 0 _or_ the flag is `true` - we may arrive to that line even if the line length is, in fact, 0 (and that's the whole point of the feature, being able to render the separator even if the status length is equal to 0).
If you prefer I could save the length in a variable to avoid computing it two times?

Reply:
> we may arrive to that line even if the line length is, in fact, 0

You're right. I missed that.

> If you prefer I could save the length in a variable to avoid computing it two times

That won't be necessary.